### PR TITLE
Paid content upload support + UX / UI improvements

### LINKED ISSUE #2210: Paid content upload support + UX / UI improvements
Overview: currently we're using tags to setup / detect purchase/rental and preorders. For now, let's skip preorders. The tag process is manual at this time, the goal here is to add something to the publish flow on all modes that allows a user to setup a purchase and/or rental. We would check for their bank account connection, and show this option, otherwise hint to setup one up to enable. We should be able to support both LBC and USD at the same time, but for now, you can grey out LBC price settings if USD is selected. 

On top of the purchase:xx.xx / rental:xx.xxx:sssssss tags, we should convert these to use the c:, and also add a separate tag for c:purchase, and c:rental. This will make it easier to filter via claim searches, since we can't do fuzzy tag searches. Both cases should hide the tags like we do with other control tags. 

The price must be in xxx.xx format, and must have a period and 2 digits. For rental, the ssss is added as seconds, but allow users to choose hours/days/months. 

Only USD and EUR is supported, and USD is treated at 1:1 currently (maybe make not if someone has non USD denom bank, not sure?). 

Others:
- [ ] add rental to claim preview if has both rental and purchase tags. Don't overwhelm the thumb, find some nicer way to show? 
- [ ] Show play button that triggers purchase flow
- [ ] mobile overflow + primary action as message: https://github.com/OdyseeTeam/odysee-frontend/issues/1906#issuecomment-1203447231
- [ ] Use startsWith for tags matching (https://github.com/OdyseeTeam/odysee-frontend/issues/2000#issuecomment-1222381268)

nice to have: 
- [ ] support embeds where you can purchase right from it / play it if you already have access

### PR DESCRIPTION
**Pull request by infinite-persistence**
Closes #2210 
Test: `kp`

## :warning: Sprint Status
https://github.com/orgs/OdyseeTeam/projects/3/views/1

## Review notes
- Reads both new and old tag formats.
- Writes new tag format.
    - The dev instance currently writes old format too for testing. Should remove after review.
- In the form, separated Fiat and SDK paywalls for now. Not sure how to link the currencies when we support beyond USD later.
- Need closer look on membership-gating, since this is the merge-conflict area.  Tried to redo some logic that handles both membership-gating and paywall-gating -- highest potential for error.

### CODE REVIEW

**Comment by tzarebczan** [comment]
- [ ] shows price free on confirmation  - I guess that's coming from the LBC price setting. Show new selections? 
image

- [ ] Odysee will be taking a 25% platform fee + standard processing fees from rentals/purchases in USD. We'll need to show that here so creators understand + link to term & conditions (I can fill in details later).  

Will get some feedback from marketing side on this flow also!

**Comment by infinite-persistence** [comment]
image

Confused with this one. Should be testing on `kp`, and the new look is this:

image

**Comment by tzarebczan** [comment]
Whoops, must have went back to Odysee on that one, sorry. Too many tabs 😂

**Comment by infinite-persistence** [comment]
> _On a related note, @toshokanneko , would be nice to find a way to show that a rental is also available in the claim preview._

Can try a variant of this, but one (or maybe just all labels) would need to be removed.  Maybe just use the icons when cannot fit?

image

**Comment by toshokanneko** [comment]
I definitely prefer this new approach without the text. However, if this is still the one that I wrote then we should probably find a solution that doesn't rely on the `screw` value....at least not without `before` or `after`.

**Comment by infinite-persistence** [comment]
> _I definitely prefer this new approach without the text. However, if this is still the one that I wrote then we should probably find a solution that doesn't rely on the screw value....at least not without before or after._

Yeah, still that version.  Agree, we'll have to re-write.  Wasted a lot time trying to it to work in various places like FilePage, etc.

**Comment by tzarebczan** [comment]
found a small bug when testing legacy functionality, would be good to fix up. On page load, we make a call to customer list with: claim id filter, to see if the user has a rental/purchase. 
image


This returns any tips on the claim too, so on the response, we have to check for type: rental or type: purchase. There also may be multiple here, so it should check them all to see if any are valid. 
```
      "is_live": null,
      "type": "tip",
      "reference_claim_id": null,
      "valid_through": null,
      "source_claim_id": "63540d14482cea0adb329240412b7f031950da52"
    }
  ]
}
```

**Comment by tzarebczan** [comment]
- [ ] would it be a lot of effort to show that content is already purchased on claim preview? Maybe run the customer/list for all results that have the tag? Would be nice to support multiple claim ids here so we can do it in one call, will check on that.

**Comment by infinite-persistence** [comment]
`--- Draft ---`
Hold off review for now, want to chance a small part...

**Comment by infinite-persistence** [comment]
- [ ] Search for `@TOM` to fill in URL and fix the breakdown formula (@tzarebczan)


`--- Review ---`
- Ok to resume review. The remaining changes will just be incremental small additions.
- Re-do the skewed price label, which now also fixes the cut-off seen in production's File Page.
    - I think there's no space to show both Rental and Purchase price, and the video duration seems nice to have still, so just show icons for that case. 
    - Don't know how to do responsive stuff yet -- can show/display based on available width when know how to.
image

File: flow-typed/gui.js [hunk 1]
```diff
@@ -3,3 +3,13 @@ declare type ListInjectedItem = {
   node: Node | (index: number, lastVisibleIndex: ?number, pageSize: ?number) => Node,
```
[the remark this answers is not in the document]
Comment (infinite-persistence) [reply flow-typed/gui.js:3]: > _Minor disalignment here (vs master)_

- The local build is missing `SIMPLE_SITE`, so the helplink "(?)" appeared and pushed it down. Ignoring for now until we support the desktop app again.
- As for the descended "5", seems to be dependent on the available serif font in the system.
    - image

[the remark this answers is not in the document]
Comment (infinite-persistence) [reply flow-typed/gui.js:3]: > _Image price upload broken?_

- Image paywall not supported, but the gray logic wasn't right. Now fixed.

[the remark this answers is not in the document]
Comment (infinite-persistence) [reply flow-typed/gui.js:3]: > _Smooth out direct claim page load? (message bubble shifts layout down, icons disappear)_

Handled disappearing icon.  Will re-think the layout shift.

File: ui/component/fileRenderInline/view.jsx [hunk 2]
```diff
@@ -12,15 +12,19 @@ type Props = {
...
+  claimIsMine: boolean,
   costInfo: any,
-  claimWasPurchased: boolean,
+  sdkPaid: boolean,
   contentRestrictedFromUser: boolean,
+  isFiatRequired: boolean,
+  isFiatPaid: ?boolean,
 };
 
 export default function FileRenderInline(props: Props) {
```
[the remark this answers is not in the document]
Comment (infinite-persistence) [reply ui/component/fileRenderInline/view.jsx:23]: Moved to https://github.com/OdyseeTeam/frontend-qa/issues/21

**Pull request merged**